projects/packages/jetpack-mu-wpcom/src/features/wpcom-blocks/code/block-front/block-front.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Find and initialize Code blocks.
+ */
+function initBlock() {
+	for ( const btn of document.querySelectorAll< HTMLButtonElement >(
+		'.a8c\\/code__btn-copy[hidden][data-copy-text]'
+	) ) {
+		btn.addEventListener(
+			'click',
+			function () {
+				// TypeScript wants the […] accessor because of an index signature.
+				// Eslint + Prettier want the dot notation.
+				// Just pick one to satisfy tooling.
+				// eslint-disable-next-line dot-notation
+				navigator.clipboard.writeText( this.dataset[ 'copyText' ]! ).catch();
+			},
+			{ passive: true }
+		);
+		btn.hidden = false;
+	}
+}
+
+if ( navigator.clipboard && ! document.body.classList.contains( 'wp-admin' ) ) {
```

Comment:
What does it make incompatible with the Interactivity API?

Reply:
The first implementation used the Interactivity API. Then I realized two things:

- I want this to work correctly on sites that don't render everything on the initial page load (o2/p2 sites, sites with infinite scroll, etc.). The Interactivity API doesn't support that, it performs hydration once on page load.
- I was using the Interactivity API to attach a single click handler. It was actually more complicated and using a library seemed quite excessive.

I decided to implement this as a JavaScript module with no dependencies.